### DISCUSSION TITLE
Question about RB-Y1 model fidelity

### DISCUSSION THREAD

**Discussion by hartikainen** [RainbowRobotics/rby1-sdk#6] [Q&A]
Hey folks,

Thanks a lot for providing the [RB-Y1 simulation models](https://github.com/RainbowRobotics/rby1-sdk/blob/5283256c045380d20909505930b6666f03dd371c/models/rby1a/model.xml)! I'm in the process of testing a (currently simulated) mobile manipulation setup with the RB-Y1 model. I wanted to ask about the fidelity of the model: how realistic are the model parameters? The bodies seem to have quite specific inertia values and realistic-looking meshes, which makes me hope that those have some truth to them. The joint and actuator parameter seem somewhat arbitrary though: for example, the [motor values](https://github.com/RainbowRobotics/rby1-sdk/blob/5283256c045380d20909505930b6666f03dd371c/models/rby1a/model.xml#L8) (particularly the `ctrlrange`) seem somewhat unorthodox to me. How accurate are these values?

Follow up questions:
  - How ready do you estimate the model to be for training e.g. reinforcement learning policies?
  - Have you all run simulated tests with the model?
  - Are there any other model-quality related things I should consider or be aware of?

**Comment by keunjun-choi** [comment]
You are correct in pointing out that parameters such as the control range or torque effort limits are not fully accurate at the moment.

Regarding the model's quality, the response can vary depending on how you define model quality. From a purely kinematic perspective, as long as you can properly align the home offsets in the real robot, the model may not differ significantly. However, from a dynamics standpoint, the model parameters (inertia values, etc.) are derived from CAD data, which may differ somewhat from reality. Even if the inertia values are reasonably accurate, there could be discrepancies due to unmodeled effects, such as friction and rotor inertia. We have successfully used this model for gravity compensation control, but keep in mind that the gear ratio is 100:1. Comparing the torque required for specific actions with the actual torque might help we evaluate the model's quality, though we haven't conducted extensive tests yet.

Additionally, the model you're using ([link](https://github.com/RainbowRobotics/rby1-sdk/blob/dev/models/rby1a/mujoco/model.xml)) employs use torque motor actuators. Therefore, you will need to integrate a position controller or a controller that combines position control with feedforward torque. For using the actuator model provided by MuJoCo, please refer to this alternative model file: [link](https://github.com/RainbowRobotics/rby1-sdk/blob/dev/models/rby1a/mujoco/model_act.xml).
